src/core/__tests__/file.test.ts

```diff
@@ -0,0 +1,88 @@
+import { MockStreamDeckClient } from '../../stream-deck/__mocks__/stream-deck-client';
+import * as __streamDeckClient from '../../stream-deck/stream-deck-client';
+import { getFileName, sanitize } from '../file';
+
+jest.mock('../../stream-deck/stream-deck-client');
+const streamDeckClient = __streamDeckClient.default as MockStreamDeckClient;
+
+const rawPath = 'C:\\fakepath\\C%3A%2FHello%2FWorld%2FFoo.txt';
+
+describe('mac', () => {
+    beforeAll(() => {
+        streamDeckClient.__connect({
+            info: {
+                application: {
+                    platform: 'mac'
+                }
+            }
+        });
+    });
+
+    /**
+     * getFileName
+     */
+    describe('getFileName', () => {
+        it('should return the file name of a sanitized path with forward slash dividers', async () => {
+            // given, when, then.
+            const fileName = await getFileName('home/Foo/Bar/Hello world.txt');
+            expect(fileName).toBe('Hello world.txt');
+        });
+
+        it('should return the path when there are no dividers', async () => {
+            // given, when, then.
+            const fileName = await getFileName('Hello world.txt');
+            expect(fileName).toBe(fileName);
+        });
+    });
+
+    /**
+     * sanitize
+     */
+    describe('sanitize', () => {
+        it('should decode and format the path with forward slashes', async () => {
+            // given, when, then.
+            const path = await sanitize(rawPath);
```

Comment:
have you tried this with a mac path instead?

Reply:
Good spot; I wonder if the fake path that Chromium adds is the same for macOS, i.e. “C:\\fakepath\\” ([ref](https://developer.elgato.com/documentation/stream-deck/sdk/property-inspector/#file)), or if it’s different. 🤔

Reply:
Going by [this reference](https://html.spec.whatwg.org/multipage/input.html#fakepath-srsly), I've updated the implementations to simply go by the path dividers present in the value; this has resulted in the removal of the async dependency which is nice! I've updated the code coverage to cater for Unix-based paths.